scala/private/phases/phase_coverage_runfiles.bzl

```diff
@@ -14,7 +14,7 @@ def phase_coverage_runfiles(ctx, p):
     if ctx.configuration.coverage_enabled and _coverage_replacements_provider.is_enabled(ctx):
```

Comment:
I'm very not familiar with the implementation, but given that this runs per-rule, it feels strange that we're not checking for `ctx.coverage_instrumented()` in this conditional and skipping the entire body (or providing explicitly empty results in an else branch) if it's not enabled.

Feel free to ignore this comment if there's complexity I'm missing.

Reply:
I may be wrong, but it still has to collect instrumented files from deps. Seems that appending `and ctx.coverage_instrumented()` produces an empty `coverage.dat`.